src/rebuild/initiator.c

```diff
@@ -792,7 +792,9 @@ rebuild_obj_ult(void *data)
 	memset(&anchor, 0, sizeof(anchor));
 	memset(&dkey_anchor, 0, sizeof(dkey_anchor));
 	memset(&akey_anchor, 0, sizeof(akey_anchor));
-	dc_obj_shard2anchor(&anchor, arg->shard);
+	dc_obj_shard2anchor(&dkey_anchor, arg->shard);
+	daos_anchor_set_flags(&dkey_anchor,
+			      DAOS_ANCHOR_FLAGS_TO_LEADER);
```

Comment:
again the question is if dtx rebuild will commit all uncommitted replicas.  is normal rebuild still only from leader?

Reply:
Fetch from leader for rebuild case is more efficient than from non-leader, because the iteration on non-leader may hit 'prepared' DTX that is new after the dtx_resync(), then the non-leader will ask the RPC sponsor (rebuild logic) to retry with leader.

Reply:
hmm, it also needs to consider load balance and performance.

Reply:
Hm, that depends on the algorithm for leader selection. If it is bad, then not only rebuild, but also the whole system during normal DAOD running may hit unbalance issue.

Reply:
I mean if there are multiple replicas, it will only fetch from leader during rebuild?

Reply:
According to our current rebuild logic, the normal IO can happen during the object rebuild processing. Then there may be new DTXs generated after the dtx_resync() the old DTX. Means that there will be 'prepared' DTX on non-leader replica. If the in-rebuild target fetch/enumerate something from non-leader, then it will get -DER_INPROGRESS failure and retry with leader. In theory rebuild logic should handle such case, but to avoid trouble, let's make the in-rebuilding target to fetch from leader only.